tests/callbacks/test_device_stats_monitor.py

```diff
@@ -69,6 +77,44 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
             for f in fields:
                 assert any(f in h for h in metrics.keys())
 
+            for f in CPU_METRIC_KEYS:
+                assert not any(f in h for h in metrics.keys()), "CPU Stats should not be included"
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=7,
+        log_every_n_steps=1,
+        accelerator="gpu",
+        devices=1,
+        callbacks=[device_stats],
+        logger=DebugLogger(tmpdir),
+        enable_checkpointing=False,
+        enable_progress_bar=False,
+    )
+
+    trainer.fit(model)
+
+
+@RunIf(max_torch="1.7")
+@RunIf(min_gpus=1)
+def test_device_stats_gpu_from_nvidia_and_cpu(tmpdir):
+    """Test GPU stats + CPU stats are logged using a logger with Pytorch < 1.8.0."""
+    model = BoringModel()
+    device_stats = DeviceStatsMonitor(cpu_stats=True)
+
+    class DebugLogger(CSVLogger):
+        @rank_zero_only
+        def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+            fields = [
+                "utilization.gpu",
+                "memory.used",
+                "memory.free",
+                "utilization.memory",
+            ] + CPU_METRIC_KEYS
+            for f in fields:
+                assert any(f in h for h in metrics.keys())
```

Comment:
```suggestion
                assert any(f in h for h in metrics)
```
also, shouldn't this be `all(...)` everywhere?